### PR TITLE
add restart for pyscf, PBC has not yet been implemented.

### PR DESCRIPTION
**Pull request by lichen5**

### CODE REVIEW

File: Manager.py [hunk 1]
```diff
@@ -18,8 +18,14 @@ def resolve_status(runner,reader,outfiles):
     if not os.path.exists(outfile):
       return 'not_started'
 
-  #We are in an error state or we haven't collected 
-  #the results. 
+  
+  #We are in an error state 
+  for outf in outfiles:
+    lines = open(outf, 'r').read().split()
+    if ('HF_done' in  lines) and ('All_done' not in lines):
```
Comment (lkwagner) [inline Manager.py:25]: This is specific to pyscf for a general function.

Looking for signals should be in the Reader object.

File: PySCF.py [hunk 2]
```diff
@@ -80,6 +80,33 @@ def is_consistent(self,other):
             .format(key,self.__dict__[key],other.__dict__[key]))
         return False
     return True
+
+  #---------------------------------------------------
+  def update_input(self, fname):
+    # check if input file already updated
+    lines = open(fname, 'r').read().split()
```
Comment (lkwagner) [inline PySCF.py:86]: Here we're editing a file. This is a rather fragile thing to do. I would prefer to rewrite the file.

File: pyscf2qwalk.py [hunk 3]
```diff
@@ -344,7 +344,8 @@ def print_cas_slater(mc,orbfile, basisfile,f, tol,fjson,root=None):
     #      + "\n#spin down orbitals \n" + beta_occ) 
     #occup.append(tmp) 
     occup.append((alpha_occ,beta_occ))
-    
+  
+  occup =[[[ int(i) for i in j] for j in k] for k in occup]
```
Comment (lkwagner) [inline pyscf2qwalk.py:349]: Can this have a bug when there is partial occupancy? Say we use smearing.

File: PySCF.py [hunk 4]
```diff
@@ -80,6 +80,33 @@ def is_consistent(self,other):
...
+      if "mc.__dict__.update(lib.chkfile.load"  in line:
+        return
+
+    # update pyscf input file for restart, load from chkfile 
+    chkfile=fname+".chkfile"
+    outlines=[]
+    with open(fname, 'r') as inf:
+      for line in inf:
+        if "print('E(HF) =',m.kernel(init_dm))" in line:
+          outlines += ["m.__dict__.update(lib.chkfile.load('%s', 'scf'))\n"%chkfile]
```
Comment (lkwagner) [inline PySCF.py:97]: Why are we updating the dict directly? The mf objects have methods that read in the check file.

**Review by lkwagner — CHANGES_REQUESTED** [review]
Hi, while this may work for your purposes, I don't think we can bring this into the codebase as it is.

File: PySCF.py (hunk 2 above)
Comment (lichen5) [reply PySCF.py:86]: Should I write it in another file with different name?

File: PySCF.py (hunk 4 above)
Comment (lichen5) [reply PySCF.py:97]: Is that pyscf.lib.chkfile.load(chkfile, key)?

Comment (armunoz) [reply PySCF.py:97]: That should give you what you want, Li.

Comment (lkwagner) [reply PySCF.py:97]: I think it should be 
m. init_guess_by_chkfile()

Using the libs directly probably will not be general.

File: PySCF.py (hunk 2 above)
Comment (lkwagner) [reply PySCF.py:86]: The PySCF writer doesn't need to know about which file to write to. The file management is not done at that level. It only needs to know that it is reading from a given check file.

This capability was already there in the pyscf writer. You need only to change the recipe to enable restarts, not the pyscf writer.

**Pull request closed without merging**